x-pack/plugins/infra/server/lib/alerting/log_threshold/log_threshold_executor.ts

```diff
@@ -101,6 +177,44 @@ const getESQuery = (
   // Negative assertions (things that "must not" match)
   const mustNotFilters = buildFiltersForCriteria(negativeCriteria);
 
+  const rangeFilters = [
+    {
+      range: {
+        [sourceConfiguration.fields.timestamp]: {
+          gte: from,
+          lte: to,
+          format: 'epoch_millis',
+        },
+      },
+    },
+  ];
+
+  const sharedSubLevelAggregations = {
+    interval_buckets: {
+      date_histogram: {
+        field: sourceConfiguration.fields.timestamp,
+        fixed_interval: interval,
+      },
+    },
+  };
+
+  const aggregations =
+    groupBy && groupBy.length > 0
+      ? {
+          groups: {
+            composite: {
+              size: 20,
```

Comment:
Out of curiosity, why did you choose this size? Is there a risk in going higher to reduce the number of pages we have to query?

Reply:
Unfortunately not. The balance is between staying below the max bucket count and response size limits on the one hand and minimizing the number of requests on the other hand. Ideally the average case would not need more than one query. So maybe 20 is realistic for grouping :shrug:.